flume-ng-core/src/main/java/org/apache/flume/tools/FlumeBeanConfigurator.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flume.tools;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+import org.apache.flume.Context;
+import org.apache.flume.conf.ConfigurationException;
+import org.slf4j.Logger;
+
+/**
+ * Utility class to enable runtime configuration of Java objects using provided
+ * Flume context objects (or equivalent). The methods use reflection to identify
+ * Fields on the configurable object and then looks for matching properties in
+ * the provided properties bundle.
+ *
+ */
+public class FlumeBeanConfigurator {
```

Comment:
I'm a little bit concerned about this FlumeBeanConfigurator class, because if it is not used with caution it could expose setters we might not would like to expose. I see why it is easier to use a util like this with the jetty configuration. We should consider moving this somewhere so only the HttpSource  could see and use it.

Also some question that could be documented is :
When the user sets both `keystorePassword` and `SslContextFactory.KeyStorePassword` which one will be set?

Reply:
Clarified the ``keystore`` bit in the docs and actually changed the order of precedence.

Regarding the ``FlumeBeanConfigurator`` I'm afraid I don't agree that we could expose anything that wasn't already exposed. The Configurator only uses public setters, so cannot set any property that wasn't already available to be set - developers obviously need to use it with their eyes open.

I've added a set of Unit tests to test the functionality of this, including testing that private setters cannot be modified. Hopefully this satisfies your concerns.